presto-native-execution/presto_cpp/main/TaskResource.cpp

```diff
@@ -329,14 +347,20 @@ proxygen::RequestHandler* TaskResource::createOrUpdateTask(
       [&](const protocol::TaskId& taskId,
           const std::string& updateJson,
           const bool summarize,
-          long startProcessCpuTime) {
-        protocol::TaskUpdateRequest updateRequest = json::parse(updateJson);
+          long startProcessCpuTime,
+          bool receiveThrift) {
+        protocol::TaskUpdateRequest updateRequest;
+        if (receiveThrift) {
+          auto thriftTaskUpdateRequest = std::make_shared<thrift::TaskUpdateRequest>();
+          thriftRead(updateJson, thriftTaskUpdateRequest);
```

Comment:
nit: maybe rename `updateJson` to something like `requestBody` (here and in other similar places)